temple/update.py

```diff
@@ -32,20 +34,16 @@ def _cookiecutter_configs_have_changed(template, old_version, new_version):
     Returns:
         bool: True if the cookiecutter.json files have been changed in the old and new versions
     """
-    temple.check.is_git_ssh_path(template)
-    repo_path = temple.utils.get_repo_path(template)
-    github_client = temple.utils.GithubClient()
-    api = '/repos/{}/contents/cookiecutter.json'.format(repo_path)
-
-    old_config_resp = github_client.get(api, params={'ref': old_version})
-    old_config_resp.raise_for_status()
-    new_config_resp = github_client.get(api, params={'ref': new_version})
-    new_config_resp.raise_for_status()
+    with tempfile.TemporaryDirectory() as clone_dir:
+        repo_dir = cc_vcs.clone(template, old_version, clone_dir)
+        old_config = json.load(open(os.path.join(repo_dir, 'cookiecutter.json')))
+        subprocess.check_call(f'git checkout {new_version}', cwd=repo_dir, shell=True, stderr=subprocess.PIPE)
+        new_config = json.load(open(os.path.join(repo_dir, 'cookiecutter.json')))
```

Comment:
I was all set to patch the mocker in test_update until this call, and then I pondered about the best way forward: a custom class with mockable methods for old and new, but that means more tests?